src/utils/viewer/ObjectPickingHelper.cpp

```diff
@@ -0,0 +1,132 @@
+#include "ObjectPickingHelper.h"
+#include <Corrade/Containers/StridedArrayView.h>
+#include <Corrade/Utility/Assert.h>
+#include <Magnum/GL/Framebuffer.h>
+#include <Magnum/GL/Renderbuffer.h>
+#include <Magnum/GL/RenderbufferFormat.h>
+#include <Magnum/Image.h>
+#include <Magnum/Magnum.h>
+#include <Magnum/PixelFormat.h>
+#include <Magnum/Shaders/Generic.h>
+#include <Magnum/Shaders/MeshVisualizer.h>
+
+namespace Cr = Corrade;
+namespace Mn = Magnum;
+using Mn::Math::Literals::operator""_rgbf;
+
+ObjectPickingHelper::ObjectPickingHelper(Mn::Vector2i viewportSize) {
+  // create the framebuffer and set the color attachment
+  recreateFramebuffer(viewportSize).mapForDraw();
```

Comment:
```suggestion
  recreateFramebuffer(viewportSize);
  mapForDraw();
```

less confusing this way I think (I would even go further and just make all methods return `void` because I don't see method chaining having any use in this context and it's just noise)

Reply:
re-creating the frame buffers and putting them immediately in use is quite natural to me. May I ask why you think method chaining here does not make sense to you?

Reply:
Because at first it looked like `recreateFramebuffer()` returns some other object (instead of `*this`), so I had to check what was the actual return type of it. 

You don't save any keystrokes with that and it only adds congitive load.